### DISCUSSION TITLE
x-read-only和readonly的区别

### DISCUSSION THREAD

**Discussion by MatriXiao88** [alibaba/formily#1632] [Q&A 2.x] [answered]
我看到这里的的[代码](https://github.com/alibaba/formily/blob/formily_next/packages/json-schema/src/transformer.ts#L226)会存在x-read-only和readonly，并且会优先使用x-read-only，那么是否两者重复了呢，因为他们最后都是被readOnly 变量使用，并没有区分对待

```js
  const readOnly = isValid(schema['x-read-only'])
    ? schema['x-read-only']
    : schema['readOnly']
```

**Comment by janryWang** [comment, marked as answer]
x-read-only是控制交互模式的，readOnly仅仅只是schema协议的校验属性，两者语意还是不太一样，虽然目前还没做啥特殊区分……
